objectbox/lib/src/box.dart

```diff
@@ -132,15 +132,20 @@ class Box<T> {
         return propVals[idPropName];
     }
 
+    _inReadTransaction(fn) {
+        Pointer<Void> txn = bindings.obx_txn_read(_store.ptr);
+        checkObxPtr(txn, "failed to created transaction");
+        var ret = fn();
+        checkObx(bindings.obx_txn_close(txn));
```

Comment:
Is there something like a finalizer (e.g. finally-block in Java or defer in Swift) that can make sure this executes, even if `fn()` throws?

Reply:
Looks like there is Finally, like in Java. https://dart.dev/guides/language/language-tour#finally

Reply:
Nice, fixed it.